src/common/ittnotify.cpp

```diff
@@ -85,22 +77,66 @@ void primitive_task_start(primitive_kind_t kind) {
     int kind_idx = (int)kind;
     assert(kind_idx >= 0);
     if (kind_idx < primitive_kind::internal_only_start) {
-        assert((size_t)kind_idx < sizeof(prim_kind_itt_strings)
-                        / sizeof(prim_kind_itt_strings[0]));
-        __itt_task_begin(itt_domain(), __itt_null, __itt_null,
-                prim_kind_itt_strings[kind_idx]);
+        assert((size_t)kind_idx < sizeof(prim_kind_itt_domain)
+                        / sizeof(prim_kind_itt_domain[0]));
+
+        return prim_kind_itt_domain[kind_idx];
     }
+
+    return prim_kind_itt_domain[0];
+}
+
+} // namespace
+
+void primitive_task_start(
+        primitive_kind_t kind, const char *pd_info, const char *primitive_op) {
+    if (kind == primitive_kind::undefined) return;
+
+    __itt_string_handle *str_handle = __itt_string_handle_create(pd_info);
+
+    __itt_task_begin(itt_domain(kind), __itt_null, __itt_null, str_handle);
```

Comment:
Could you confirm that there is no extra overhead from using different domain for each primitive kind?
I believe we tried that in the past ([see this commit](https://github.com/uxlfoundation/oneDNN/commit/557f3f0ba9e366fac082427fff9d149eb0f75fce)), but we had to revert as this caused performance regressions, even when vtune collection was not ongoing.